### PR TITLE
fix: fix asha max concurrent trials

### PR DESCRIPTION
**Pull request by stoksc**
## Description
This change fix and bug and makes a shortcoming of the current `max_concurrent_trials` implementation  explicit in the documenation.

## Commentary (optional)
The long term fix is much more complicated and probably involves redesigning the searchers a bit.

## Checklist

- [x] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: master/pkg/searcher/adaptive_asha.go [hunk 1]
```diff
@@ -44,6 +44,9 @@ func getBracketMaxConcurrentTrials(
 	remainder := 0
 	numBrackets := len(maxTrials)
 	bracketMaxConcurrentTrials := make([]int, 0, numBrackets)
+	// Without this, the remainder will be less than numBrackets and later brackets will,
```
Comment (stoksc) [inline master/pkg/searcher/adaptive_asha.go:47]:
```suggestion
	// Without this, the remainder will be less than numBrackets and later brackets will
```

File: master/pkg/searcher/adaptive_asha.go [hunk 2]
```diff
@@ -44,6 +44,9 @@ func getBracketMaxConcurrentTrials(
...
+	// not receive a constraint on bracketMaxConcurrentTrials.
+	maxConcurrentTrials = min(maxConcurrentTrials, numBrackets)
```
Comment (liamcli) [inline master/pkg/searcher/adaptive_asha.go:49]: i don't think this fixes it.  we allow users to pass in 0 if they want a reasonable default for max_concurrent_trials.  
i think we should do:
maxConcurrentTrials = max(numBrackets, maxConcurrentTrials) after line 49 
before we do
minTrials = maxConcurrentTrials / numBrackets

Comment (stoksc) [reply master/pkg/searcher/adaptive_asha.go:49]: 🤦

Comment (stoksc) [reply master/pkg/searcher/adaptive_asha.go:49]: brain broke, thanks @liamcli

File: master/pkg/searcher/adaptive_asha.go [hunk 3]
```diff
@@ -47,6 +47,9 @@ func getBracketMaxConcurrentTrials(
 	if maxConcurrentTrials == 0 {
 		minTrials = max(maxTrials[numBrackets-1], int(divisor))
 	} else {
+		// Without this, the remainder will be less than numBrackets and later brackets will,
+		// not receive a constraint on bracketMaxConcurrentTrials.
+		maxConcurrentTrials = min(maxConcurrentTrials, numBrackets)
```
Comment (liamcli) [inline master/pkg/searcher/adaptive_asha.go:52]: max

**Review by liamcli — APPROVED** [review]
LGTM.

nit: do we need the changes to mnist_pytorch/adaptive.yaml and tools/agent.yaml?

File: docs/release-notes/1719-fix-asha-max-concurrent-trials.txt [hunk 4]
```diff
@@ -0,0 +1,6 @@
+:orphan:
+
+**BUG FIXES**
+
+-  HP Search: Fix a bug where ``adaptive_asha`` could run higher max
```
Comment (stoksc) [inline docs/release-notes/1719-fix-asha-max-concurrent-trials.txt:5]:
```suggestion
-  HP Search: Fix a bug where ``adaptive_asha`` could run with more max
```

**Pull request merged**